tests/conftest.py

```diff
@@ -81,13 +76,13 @@ def args(
 
         extra_args.extend(("--enable-lora", f"--lora-modules={name}={path}"))
 
-    monkeysession.setattr(
+    monkeypatch.setattr(
         sys,
         "argv",
         [
             "__main__.py",
-            f"--grpc-port={grpc_server_thread_port}",
-            f"--port={http_server_thread_port}",
+            f"--grpc-port={grpc_server_port}",
+            f"--port={http_server_port}",
```

Comment:
Should we add `"--disable-frontend-multiprocessing"` for now? If there is an exception in the initialization of the server, the tests hang forever even with the timeout.

I noticed this because I sometimes set `NUM_GPUS=0`  and this causes a division by 0 division by 0 error in our tgis adapter code. (But this error is out of scope of this discussion)

Reply:
FYI: https://github.com/vllm-project/vllm/issues/7213

Reply:
I cleaned up the tests code a bit, so you might not see that hang anymore.

Regarding `--disable-frontend-multiprocessing`: I parametrized tests so that we test both enabled and disabled.

Tests pass in both cases, although there's an annoying bug at tests cleanup which causes pytest to exit with a non-zero exit code even with all tests passing. I ended up temporarily disabling that test so that we can get this merged asap.

Reply:
FYI, https://github.com/vllm-project/vllm/pull/7217 fixes this particular hang.